pkg/arvo/sys/vane/khan.hoon

```diff
@@ -0,0 +1,230 @@
+              $>(%deal task:gall)                       ::  full transmission
+          ==                                            ::
+          $:  %k                                        ::    to self
+              $>(%fard task)                            ::  internal thread
+      ==  ==                                            ::
+    +$  sign                                            ::    in response $<-
+      $%  $:  %gall                                     ::    from %gall
+              $>(%unto gift:gall)                       ::  update
+          ==                                            ::
+          $:  %khan                                     ::    from self
+              $>  ?(%arow %avow)  gift                  ::  thread result
+      ==  ==                                            ::
+    +$  khan-state                                      ::
+      [%0 hey=duct]                                     ::  current unix duct
+    --                                                  ::
+=>
+|%
+++  get-beak
+  |=  [=bear now=@da]
+  ?@(bear [our bear %da now] bear)
+++  get-dais
+  |=  [=beak =mark rof=roof]
+  ^-  dais:clay
+  ?~  ret=(rof ~ %cb beak /[mark])
+    ~|(mark-unknown+mark !!)
+  ?~  u.ret
+    ~|(mark-invalid+mark !!)
+  ?>  =(%dais p.u.u.ret)
+  !<(dais:clay q.u.u.ret)
+++  get-tube
+  |=  [=beak =mark =out=mark rof=roof]
+  ^-  tube:clay
+  ?~  ret=(rof ~ %cc beak /[mark]/[out-mark])
+    ~|(tube-unknown+[mark out-mark] !!)
+  ?~  u.ret
+    ~|(tube-invalid+[mark out-mark] !!)
+  ?>  =(%tube p.u.u.ret)
+  !<(tube:clay q.u.u.ret)
+++  make-wire
+  |=  [=beak =mark]
+  ^-  wire
+  :~  %fyrd
+      (scot %p p.beak)
+      q.beak
+      -.r.beak
+      (scot r.beak)
+      mark
+  ==
+++  read-case
+  |=  a=[p=@ @]
+  ^-  case
+  ::  XX  should be:
+  ::    ?>  ?=(?(%da %tas %ud) p.a)
+  ::    [p.a (slav a)]
+  ::
+  ?+  p.a   !!
+    %da   [p.a (slav a)]
+    %tas  [p.a (slav a)]
+    %ud   [p.a (slav a)]
+  ==
+++  read-wire
+  |=  =wire
+  ^-  (pair beak mark)
+  ~|  khan-read-wire+wire
+  ?>  ?=([%fyrd @ @ @ @ @ ~] wire)
+  :_  i.t.t.t.t.t.wire
+  :+  (slav %p i.t.wire)
+    i.t.t.wire
+  (read-case [i i.t]:t.t.t.wire)
+++  start-spider
+  |=  [our=@p =vase]
+  ^-  note
+  [%g %deal [our our] %spider %poke %spider-start vase]
+++  watch-spider
+  |=  [our=@p =path]
+  ^-  note
+  [%g %deal [our our] %spider %watch path]
+--
+=|  khan-state
+=*  state  -
+|=  [now=@da eny=@uvJ rof=roof]
+=*  khan-gate  .
+^?
+|%
+::  +call: handle a +task request
+::
+++  call
+  |=  $:  hen=duct
+          dud=(unit goof)
+          wrapped-task=(hobo task)
+      ==
+  ^-  [(list move) _khan-gate]
+  ::
+  =/  =task  ((harden task) wrapped-task)
+  ?^  dud
+    ~|(%khan-call-dud (mean tang.u.dud))
+  ?+    -.task  [~ khan-gate]
+      %born
+    [~ khan-gate(hey hen)]
+  ::
+      %fard
+    =/  tid=@ta
+      (cat 3 'khan-fyrd--' (scot %uv (sham eny)))
```

Comment:
Arvo does not currently rotate entropy during an event, so thread IDs will collide if multiple `%fard`'s are passed in a single event. So ID generation will need to be stateful within `%kahn`.

Reply:
resolved this, but not sure yet how to test it — just call `+call` multiple times against the same core?

Reply:
ah, you just call multiple times against the same `eny` argument. ok, working on this